drivers/overlay/joinleave.go

```diff
@@ -0,0 +1,90 @@
+package overlay
+
+import (
+	"fmt"
+
+	"github.com/docker/libnetwork/driverapi"
+	"github.com/docker/libnetwork/types"
+	"github.com/vishvananda/netlink"
+)
+
+// Join method is invoked when a Sandbox is attached to an endpoint.
+func (d *driver) Join(nid, eid types.UUID, sboxKey string, jinfo driverapi.JoinInfo, options map[string]interface{}) error {
+	if err := validateID(nid, eid); err != nil {
+		return err
+	}
+
+	n := d.network(nid)
+	if n == nil {
+		return fmt.Errorf("could not find network with id %s", nid)
+	}
+
+	ep := n.endpoint(eid)
+	if ep == nil {
+		return fmt.Errorf("could not find endpoint with id %s", eid)
+	}
+
+	var err error
+	n.Do(func() { err = d.initSandbox(n) })
+	if err != nil {
+		return fmt.Errorf("network sandbox initialization failed: %v",
+			err)
+	}
+
+	sbox := n.sandbox()
+
+	name1, name2, err := createVethPair()
+	if err != nil {
+		return err
+	}
+
+	if err := sbox.AddInterface(name1, "veth",
+		sbox.InterfaceOptions().Master("bridge1")); err != nil {
```

Comment:
am seeing a panic when attaching a sandbox to the only endpoint published on a overlay network. Need to dig further.

```
runtime error: invalid memory address or nil pointer dereference
goroutine 116 [running]:
net/http.func·011()
	/usr/local/go/src/net/http/server.go:1130 +0xbb
github.com/docker/libnetwork/drivers/overlay.(*driver).Join(0xc208052bd0, 0xc2080dd840, 0x40, 0xc20819bb80, 0x40, 0xc2081ee860, 0x19, 0x7fdb481ba0f8, 0xc208052d80, 0x0, ...)
	/home/vagrant/workspace/src/github.com/docker/libnetwork/drivers/overlay/joinleave.go:42 +0x5e7
github.com/docker/libnetwork.(*endpoint).Join(0xc208052d80, 0xc2081f7080, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/vagrant/workspace/src/github.com/docker/libnetwork/endpoint.go:402 +0x7a3
```